### DISCUSSION TITLE
Viper is deleting all comments when viper.Set and viper.WriteConfig is done

### DISCUSSION THREAD

**Discussion by AlexMcGopher** [spf13/viper#1173] [Q&A] [answered]
Viper is deleting in my example all comments when viper.Set and viper.WriteConfig is done.

I read in the values of a config file (yaml type) and change the value of one of the Parameters.
As I have some comments in the config file, they are all gone after the write and the values are ordered by name A-Z,
empty lines are also gone....

config.yml before write:
```
# Insert IP addr here:
Instance:
  Local_IP : "10.0.55.1"
# residence:
address: "5th Avenue"
```

config.yml after viper.Set and viper.WriteConfig: comments lost, empty lines and other order
```
address: "10th Avenue"
Instance:
  Local_IP : "10.0.55.8"
```

expected behaviour:
```
# Insert IP addr here:
Instance:
  Local_IP : "10.0.55.8"
# residence:
address: "10th Avenue"
```

Is there any possibility to keep the file as it has been before and just change the Values?

Thanks and best regards, Alex

**Comment by sagikazarmark** [comment, marked as answer]
Unfortunately, no. It would be quite hard to implement with very minimal benefit.

WriteConfig is not supposed to be used for updating config files. It's more like a tool to dump default configuration.

**Reply by AlexMcGopher** [reply]
Thanks for quick answer !
